scripts/components/Repo.js

```diff
@@ -1,19 +1,24 @@
 'use strict';
 
 import React, { PropTypes } from 'react';
-import PureRenderMixin from 'react/lib/ReactComponentWithPureRenderMixin';
+import shallowEqual from 'react/lib/shallowEqual';
 import { Link } from 'react-router';
 
-const Repo = React.createClass({
-  mixins: [PureRenderMixin],
+class Repo extends React.Component {
 
-  propTypes: {
-    repo: PropTypes.object.isRequired,
-    owner: PropTypes.object.isRequired
-  },
+  // TODO: make it either a HigherOrderComponent
```

Comment:
It works if you don't use ES6 function arrow, as babel will compile `this` to `undefined`

```js
// this works
export default function shouldComponentUpdate (nextProps, nextState) {
  return !shallowEqual(this.props, nextProps) || !shallowEqual(this.state, nextState)
}

// this doesn't
export default (nextProps, nextState) => {
  return !shallowEqual(this.props, nextProps) || !shallowEqual(this.state, nextState)
}
```

Babel compilation

```js
const getBar1 = function () { return this.bar }
const getBar2 = () => { return this.bar }

// =>
"use strict";

var getBar1 = function getBar1() {
  return this.bar;
};
var getBar2 = function getBar2() {
  return undefined.bar;
};
```